src/lib.rs

```diff
@@ -103,18 +105,58 @@ pub fn collect_stars(config: Config) -> Result<(), Box<dyn error::Error>> {
     if let Some(ref token) = config.token {
         builder.set_authorization_token(token.to_owned());
     }
+    else {
+        println!("Authentication Warning: This is an unauthenticated request with a limit of 60 requests per hour. Re-run this program using an auth token by adding `--token <auth-token>` for an increased quota of 5000 requests per hour.")
+    }
 
     let client = builder.build()?;
 
     let mut stars: Vec<Star> = Vec::new();
 
     let mut next_link = config.url();
 
+    let total_rate_limit = "X-RateLimit-Limit";
+    let rate_limit_remaining = "X-RateLimit-Remaining";
+    let rate_limit_reset_time = "X-RateLimit-Reset";   
+    
+    let mut remaining: i32 = 0; 
+    let mut total: i32 = 0;
+    let mut timestamp_str = String::new();
     while next_link.is_some() {
         if let Some(link) = next_link {
             let mut res = client.get(&link).send()?;
-            next_link = extract_link_next(res.headers());
 
+            for header in res.headers().iter() {
+                if header.name() == total_rate_limit {
+                    total = header.value_string().parse::<i32>().unwrap();
+                }
+                if header.name() == rate_limit_remaining {
+                    remaining = header.value_string().parse::<i32>().unwrap();
+                }
+                if header.name() == rate_limit_reset_time {
+                        let reset_time = header.value_string();
+                        let timestamp = reset_time.parse::<u64>().unwrap();
+
+                        // Creates a new SystemTime from the specified number of whole seconds
+                        let d = UNIX_EPOCH + Duration::from_secs(timestamp);
+                        
+                        // Create DateTime from SystemTime
+                        let datetime = DateTime::<Local>::from(d);
+
+                        // Formats the combined date and time with the specified format string.
+                        timestamp_str = datetime.format("%Y-%m-%d %I:%M").to_string();
```

Comment:
Not totally sold on the date format here. In my test I ended up with `2018-11-27 11:13`. `%I` gives you the 12-hour hour number, but without `AM` or `PM` it's not very useful.

Since the request limit generally resets every hour, I think it would be more useful to tell the user a number of minutes (or hours, potentially) until their limit resets, rather than a full date, e.g. `"Your request limit will reset in 24 minutes"`.